src/relay/op/tensor/reduce.cc

```diff
@@ -58,17 +59,20 @@ inline std::vector<int64_t> GetReduceAxes(const uint32_t indim,
 
   std::vector<int64_t> in_axes;
   for (auto i : inaxis) {
-    auto k = as_const_int(i);
+    const int64_t* k = as_const_int(i);
     CHECK(k != nullptr) << "Reduce axis need to be constant, cannot be symbolic";
```

Comment:
`axis = k[0]; `cause segmentation fault, if no check. You mean define `in_axes` as `std::vector<IndexExpr>` and avoid all bound checks if axis is symbolic?

Reply:
if conversion to integer fails, we can break from the loop, and mark a check_bound flag to be false

Reply:
Thanks a lot for the clarification and your time. 
To be more clear, you mean change return type of `GetReduceAxes` to `std::vector<IndexExpr> `  similar to [commit](https://github.com/dmlc/tvm/pull/1865/commits/7c0256fb7c7353c5097028ef0c21b01f3d54308f
) before and along with this, do the check if axis is integer?

Without changing prototype, in current code, if conversion fails, there is no point in continuing. `GetReduceAxes` wont be able to return any value.

Reply:
Sorry, I was confused, the current code is fine to assume the axis is a constant integer. What I meant is that we do not need the shape production of the axis to be symbolic

Reply:
Thanks again and  sorry for the confusions. 

```
  std::vector<int64_t> in_axes;
  int64_t axes_pdt = 1;
  for (auto i : inaxis) {
    const int64_t* k = as_const_int(i);
    CHECK(k != nullptr) << "Reduce axis need to be constant, cannot be symbolic";
    int64_t axis = k[0];
    if (axis < 0) {
      axis = axis + indim;
    }
    axes_pdt *= axis;
    // Check out of bounds error
    CHECK(axes_pdt < std::numeric_limits<int32_t>::max())
      << "Axes product cannot be greater than int32 max.";
    CHECK(axis >= 0)
      << "Axis out of bounds in reduce operator.";
    CHECK(axis < indim)
      << "Axis out of bounds in reduce operator.";
    in_axes.push_back(axis);
  }
```

> if conversion to integer fails, we can break from the loop, and mark a check_bound flag to be false

This i have not done, as the shape product wont be symbolic here because its derived after as_const_int conversion and we cannot continue the function without integer values.

Please review the above code snippet, anything else i need to add.